components/brave_wallet_ui/page/screens/send/send/send.tsx

```diff
@@ -0,0 +1,336 @@
+    }
+
+    const amountWei = new Amount(sendAmount).multiplyByDecimals(
+      selectedSendAsset.decimals
+    )
+
+    if (amountWei.isZero()) {
+      return false
+    }
+
+    return amountWei.gt(sendAssetBalance)
+  }, [sendAssetBalance, sendAmount, selectedSendAsset])
+
+  const sendAmountFiatValue = React.useMemo(() => {
+    if (!selectedSendAsset || sendAssetBalance === '' || selectedSendOption === 'nft') {
+      return ''
+    }
+    return computeFiatAmount(spotPrices, {
+      decimals: selectedSendAsset.decimals,
+      symbol: selectedSendAsset.symbol,
+      value: new Amount(sendAmount !== '' ? sendAmount : '0')
+        .multiplyByDecimals(selectedSendAsset.decimals) // ETH → Wei conversion
+        .toHex()
+    }).formatAsFiat(defaultCurrencies.fiat)
+  }, [spotPrices, selectedSendAsset, sendAmount, defaultCurrencies.fiat, sendAssetBalance, selectedSendOption])
+
+  const reviewButtonText = React.useMemo(() => {
+    return sendAmountValidationError
+      ? getLocale('braveWalletDecimalPlacesError')
+      : insufficientFundsError
+        ? getLocale('braveWalletNotEnoughFunds')
+        : (addressError !== undefined && addressError !== '')
+          ? addressError
+          : (addressWarning !== undefined && addressWarning !== '')
+            ? addressWarning
+            : getLocale('braveWalletReviewOrder')
+  }, [insufficientFundsError, addressError, addressWarning, sendAmountValidationError])
+
+  const isReviewButtonDisabled = React.useMemo(() => {
+    return toAddressOrUrl === '' ||
+      parseFloat(sendAmount) === 0 ||
+      sendAmount === '' ||
+      insufficientFundsError ||
+      (addressError !== undefined && addressError !== '') ||
+      sendAmountValidationError !== undefined
+  }, [toAddressOrUrl, sendAmount, insufficientFundsError, addressError, sendAmountValidationError])
+
+  const selectedTokensNetwork = React.useMemo(() => {
+    if (selectedSendAsset) {
+      return getTokensNetwork(networks, selectedSendAsset)
+    }
+    return undefined
+  }, [selectedSendAsset, networks])
+
+  const showResolvedDomainAddress = React.useMemo(() => {
+    if (
+      (addressError === undefined || addressError === '') &&
+      toAddress &&
+      endsWithAny(allSupportedExtensions, toAddressOrUrl.toLowerCase())
+    ) {
+      setFoundAddressPosition(document.getElementById(INPUT_WIDTH_ID)?.clientWidth ?? 0)
+      return true
+    }
+    return false
+  }, [addressError, toAddress, toAddressOrUrl])
+
+  // Effects
+  React.useEffect(() => {
+    // Keeps track of the Swap Containers Height to update
+    // the network backgrounds height.
+    setBackgroundHeight(ref?.current?.clientHeight ?? 0)
+  }, [ref])
+
+  React.useEffect(() => {
+    // Changes network background opacity to 0.6 after changing networks
+    setBackgroundOpacity(0.6)
+    // Changes network background opacity back to 0.3 after 1 second
+    setTimeout(() => setBackgroundOpacity(0.3), 1000)
+  }, [selectedTokensNetwork])
```

Comment:
Need to prevent setting state after unmount

```suggestion
  React.useEffect(() => {
    const subscribed = true
    // Changes network background opacity to 0.6 after changing networks
    setBackgroundOpacity(0.6)
    // Changes network background opacity back to 0.3 after 1 second
    setTimeout(() => {
      if (subscribed) {
        setBackgroundOpacity(0.3)
      }
    }, 1000)
    
    // cleanup
    return () => {
      subscribed = false
    }
  }, [selectedTokensNetwork])
```

Reply:
Updated!